.github/workflows/windows.yml

```diff
@@ -1,71 +1,95 @@
 ---
-name: windows-latest
+name: windows
 
 on: [push, pull_request]
+
 env:
   ARCH: "x64"
-  VC: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Auxiliary\\Build\\vcvarsall.bat"
+  VC2019: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Auxiliary\\Build\\vcvarsall.bat"
+  VC2016: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Enterprise\\VC\\Auxiliary\\Build\\vcvarsall.bat"
+
 jobs:
   compile: # name of the job. Jobs run in parallel unless specified otherwise.
     name: compile
-    runs-on: windows-latest
+    runs-on: ${{ matrix.windows-version }}
     strategy:
       matrix:
-        python-version: ['3.6']
-        build-version: ["release", "debug"]
+        windows-version: [windows-2016, windows-2019]
+        python-version: [3.6]
+        build-version: [release, debug]
     steps: # each - is a new sequentially run step
       - uses: actions/checkout@master
 
       - name: Setup python
         uses: actions/setup-python@v1
         with:
           python-version: ${{ matrix.python-version }}
-          architecture: ${{ env.ARCH }}
 
       - name: compile
         shell: cmd
+        env:
+          CC: cl
+          CXX: cl
         run: |
-          "${{ env.VC }}" ${{ env.ARCH }} & python build.py ${{ matrix.build-version }}
+          IF "${{ matrix.windows-version }}" == "windows-2016" (
+            "${{ env.VC2016 }}" ${{ env.ARCH }} & python build.py ${{ matrix.build-version }}
+          )
+          IF "${{ matrix.windows-version }}" == "windows-2019" (
+            "${{ env.VC2019 }}" ${{ env.ARCH }} & python build.py ${{ matrix.build-version }}
+          )
```

Comment:
I take a look, if we can improve this, because, I have to call those scripts frequently in my issue

Reply:
Found a nicer way. See issue1005. If you want you can change it, but I do not require it, as I will change it soon after :)